src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Dependency.cs

```diff
@@ -368,11 +368,10 @@ public static string GetID(ITargetFramework targetFramework, string providerType
             }
             finally
             {
-                sb.Clear();
-
                 // Prevent holding on to large builders
-                if (sb.Length < 1000)
+                if (sb?.Capacity < 1000)
```

Comment:
Whoops - my fault.

Reply:
It would have been hard to hit this in practice. Perhaps if `targetFramework.ShortName` was `null`.